dev/releases/update_website.py

```diff
@@ -14,416 +14,193 @@
-                   help="git remote to which --branch is pushed (default: origin)")
-group.add_argument("--pr-remote", type=str, default="origin", # TODO
-                   help="git remote to which a PR is made (default: origin)")
-# --gap-fork is essentially just a feature for testing
+group.add_argument("--token", type=str, help="GitHub access token")
 group.add_argument("--gap-fork", type=str, default="gap-system",
-                   help="GitHub GAP fork to search for releases (default: gap-system)")
-
+        help="GitHub GAP fork to search for releases (for testing; default: gap-system)")
 args = parser.parse_args()
 
+utils.verify_command_available("git")
+utils.verify_git_repo()
+utils.verify_git_clean()
 
-################################################################################
-# Verify that commands are available
-verify_command_available("git")
-# Verify that the pwd is a clean git repo
-verify_git_repo()
-verify_git_clean()
-
-
-################################################################################
-# Validate the arguments
-
-if args.tag != None and not is_possible_gap4_release_tag(args.tag):
-    error("unrecognised format for GAP release tag: " + args.tag)
-
-if args.use_github_date and args.date:
-    error("--use-github-date and --date/-d were both specified; use at most one")
-elif args.date:
-    try:
-        release_date = datetime.datetime.strptime(args.date, "%Y-%m-%d")
-    except ValueError:
-        error("--date YYYY-MM-DD")
-
-if (args.push_remote != None and len(args.push_remote) == 0) or (args.pr_remote != None and len(args.pr_remote) == 0):
-    error("--push-remote and --pr-remote cannot be empty strings")
-
-if args.tmpdir != None:
-    if not os.path.isdir(args.tmpdir) or not os.access(args.tmpdir, os.W_OK):
-        error("--tmpdir does not describe a writeable directory")
-    tmpdir = args.tmpdir
-else:
-    tmpdir = tempfile.gettempdir()
-if not tmpdir[-1] == "/":
-    tmpdir += "/"
+pwd = os.getcwd()
+tmpdir = tempfile.gettempdir()
 
 
-################################################################################
-# Use the GitHub API to find the appropriate GAP release on GitHub
-utils.CURRENT_REPO_NAME = os.environ.get("GITHUB_REPOSITORY", args.gap_fork + "/gap")
-utils.initialize_github(args.token)
-if args.tag:
-    try:
-        release = utils.CURRENT_REPO.get_release(args.tag)
-    except:
-        error("non-existent or ambiguous release tag name: " + args.tag)
-else:
-    # Using latest release; therefore ignore drafts and prerelease
-    release = [ x for x in utils.CURRENT_REPO.get_releases() if \
-                not x.draft and not x.prerelease and is_possible_gap4_release_tag(x.tag_name)]
+# When we find a previously unknown release, we will extract the release date
+# from the configure.ac file contained in gap-{version}-core.tar.gz.
+# This date is set by the make_archives.py script.
+def get_date_from_configure_ac():
```

Comment:
This function does more than this comment and its name let on: it in fact downloads and extracts a tarball, which I found surprising.

I would suggest to at least update the comment. On the long run, it might be better to split this into two (or more) functions, as the fragments tend to be easier to work with / compose differently (e.g. the downloaded 
tarball resp. the extracted content, may be useful for other things. E.g. at some point, we will also want to automate updating the documentation on the website)

But if it works now, I don't want to ask you to break it now.

Reply:
I will at least update the comment, and I agree that splitting it would be better.

Reply:
I've split this up.